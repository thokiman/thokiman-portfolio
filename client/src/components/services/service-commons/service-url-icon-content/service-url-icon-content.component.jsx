import React from 'react';
import {
  ServiceUrlIcon,
  ServiceUrlIconContainer,
  ServiceUrlIconItem,
} from './service-url-icon-content.styles';

export const ServiceUrlIconContent = ({ items }) => {
  return (
    <ServiceUrlIcon>
      {items.map(({ id, linkUrl, iconPath: { field, iconPath } }) => {
        return (
          <ServiceUrlIconContainer
            className='service-url-icon'
            key={id}
            to={{ pathname: `${linkUrl}` }}
            target='_blank'
          >
            <ServiceUrlIconItem src={iconPath} alt={field} />
          </ServiceUrlIconContainer>
        );
      })}
    </ServiceUrlIcon>
  );
};

export default ServiceUrlIconContent;
