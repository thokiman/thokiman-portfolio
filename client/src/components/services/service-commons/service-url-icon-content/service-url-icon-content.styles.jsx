import { Link } from 'react-router-dom';
import styled, { keyframes } from 'styled-components';

const blueColor = '#19dcea';
const purpleColor = '#b22cff';
const redColor = '#ea2222';
const yellowColor = '#f5be10';
const greenColor = '#3bd80d';

const maxWidthSmartphone = 800;

export const ServiceUrlIcon = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 10px 20px;
  width: 100%;
  justify-content: center;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    display: flex;
    flex-direction: column;
    align-items: center;
  }
`;

const serviceUrlIconContainer = keyframes`
                 0% {
                     background: ${blueColor};
                     opacity: 0.6;
                 }

                 25% {
                     background: ${purpleColor};
                     opacity: 0.8;
                 }

                 50% {
                     background: ${redColor};
                     opacity: 0.7;
                 }

                 75% {
                     background:${yellowColor} ;
                     opacity: 0.9;
                 }

                 100% {
                     background: ${greenColor};
                     opacity: 1;
                 }
`;
export const ServiceUrlIconContainer = styled(Link)`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;

  cursor: pointer;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 42vw;
    height: 22vh;
  }
  @media only screen and (max-width: 550px) {
    width: 35vw;
    height: 18vh;
  }
  @media only screen and (max-width: 470px) and (max-height: 823px) {
    width: 37vw;
    height: 10vh;
  }

  @media only screen and (max-width: 470px) and (max-height: 812px) {
    width: 35vw;
  }
  @media only screen and (max-width: 470px) and (max-height: 736px) {
    width: 40vw;
    height: 14vh;
  }
  @media only screen and (max-width: 470px) and (max-height: 667px) {
  }
  @media only screen and (max-width: 350px) and (max-height: 568px) {
  }

  @media only screen and (max-width: 300px) {
    width: 44vw;
    height: 16vh;
  }
  &:hover {
    animation: ${serviceUrlIconContainer} 8s linear infinite alternate both;
  }
`;

export const ServiceUrlIconItem = styled.img`
  width: 80%;
  height: 80%;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 100%;
    height: 90%;
  }
`;
ServiceUrlIcon.displayName = 'ServiceUrlIcon';
ServiceUrlIconContainer.displayName = 'ServiceUrlIconContainer';
ServiceUrlIconItem.displayName = 'ServiceUrlIconItem';
