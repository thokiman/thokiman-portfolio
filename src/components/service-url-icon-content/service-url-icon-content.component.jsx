import React from "react";
import { Link } from "react-router-dom";

import "./service-url-icon-content.styles.scss";

const ServiceUrlIconContent = ({ items }) => {
  return (
    <div className="service-url-icon">
      {items.map(({ id, linkUrl, iconPath: { field, iconPath } }) => {
        return (
          <Link
            key={id}
            to={{ pathname: `${linkUrl}` }}
            target="_blank"
            className="service-url-icon-container"
          >
            <img src={iconPath} alt={field} className="service-url-icon-item" />
          </Link>
        );
      })}
    </div>
  );
};

export default ServiceUrlIconContent;
