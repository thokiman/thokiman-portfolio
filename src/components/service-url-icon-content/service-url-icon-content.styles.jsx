import { Link } from "react-router-dom";
import styled, { keyframes } from "styled-components";

export const ServiceUrlIcon = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 10px 20px;
  width: 100%;
  justify-content: center;
`;

const serviceUrlIconContainer = keyframes`
                 0% {
                     background: #19dcea;
                     opacity: 0.6;
                 }

                 25% {
                     background: #b22cff;
                     opacity: 0.8;
                 }

                 50% {
                     background: #ea2222;
                     opacity: 0.7;
                 }

                 75% {
                     background: #f5be10;
                     opacity: 0.9;
                 }

                 100% {
                     background: #3bd80d;
                     opacity: 1;
                 }
`;
export const ServiceUrlIconContainer = styled(Link)`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;

  cursor: pointer;
  &:hover {
    animation: ${serviceUrlIconContainer} 8s linear infinite alternate both;
  }
`;

export const ServiceUrlIconItem = styled.img`
  width: 80%;
  height: 80%;
`;
ServiceUrlIcon.displayName = "ServiceUrlIcon";
ServiceUrlIconContainer.displayName = "ServiceUrlIconContainer";
ServiceUrlIconItem.displayName = "ServiceUrlIconItem";
