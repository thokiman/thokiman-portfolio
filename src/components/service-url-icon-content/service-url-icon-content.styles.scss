 .service-url-icon {
     display: grid;
     grid-template-columns: repeat(3, 1fr);
     gap: 10px 20px;
     width: 100%;
     justify-content: center;


     .service-url-icon-container {

         display: flex;
         justify-content: center;
         align-items: center;
         width: 100%;
         height: 100%;


         cursor: pointer;

         &:hover {

             animation: color-change-5x 8s linear infinite alternate both;

             @keyframes color-change-5x {
                 0% {
                     background: #19dcea;
                     opacity: 0.6;
                 }

                 25% {
                     background: #b22cff;
                     opacity: 0.8;
                 }

                 50% {
                     background: #ea2222;
                     opacity: 0.7;
                 }

                 75% {
                     background: #f5be10;
                     opacity: 0.9;
                 }

                 100% {
                     background: #3bd80d;
                     opacity: 1;
                 }
             }
         }

         .service-url-icon-item {
             width: 80%;
             height: 80%;

         }
     }
 }