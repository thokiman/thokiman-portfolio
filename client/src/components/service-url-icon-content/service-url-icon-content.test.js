import React from "react";
import { shallow } from "enzyme";

import { ServiceUrlIconContent } from "./service-url-icon-content.component";

describe("ServiceUrlIconContent component", () => {
  let wrapper;
  const mockArray = [];

  beforeEach(() => {
    const mockProps = {
      items: mockArray,
    };
    wrapper = shallow(<ServiceUrlIconContent {...mockProps} />);
  });

  it("should render ServiceUrlIconContent component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
