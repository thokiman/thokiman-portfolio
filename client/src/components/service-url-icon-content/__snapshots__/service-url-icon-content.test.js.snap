// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceUrlIconContent component should render ServiceUrlIconContent component 1`] = `ShallowWrapper {}`;
